api/src/main/java/org/hawkular/inventory/api/WriteRelationshipInterface.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.api;
+
+import org.hawkular.inventory.api.model.Relationship;
+
+/**
+ * Generic methods to write access to relationships.
+ *
+ * @param <Entity> the entity type
+ * @param <Blueprint> the blueprint type that supplies data necessary to create a new relationship
+ * @param <Single> the access interface to a single relationship
+ *
+ * @author Jirka Kremser
+ * @since 1.0
+ */
+interface WriteRelationshipInterface<Entity, Blueprint, Single> {
+
+    /**
+     * Creates a new relationship at the current position in the inventory traversal.
+     *
+     * @param blueprint the blueprint to be used to create the new relationship
+     * @return access interface to the freshly created relationship
+     *
+     * @throws org.hawkular.inventory.api.RelationNotFoundException if the relationship already exists
+     */
+//    Single create(Blueprint blueprint) throws RelationNotFoundException;
```

Comment:
IMO, the signature doesn't make sense. I commented it out because I wanted to discuss it before coding.

Imagine you navigate to some place in the graph, like
```java
`Relationships.ReadWrite rw = inventory.tenants().get("foo").relationships();`
```
..and now you want to create a new relationship
```java
Relationship.Blueprint rel = new Relationship.Blueprint("name", watSource, someTarget);
rw.create(rel);
```

It's strange, because I can specify arbitrary source in the blueprint and it should respect the path I am currently in. That was my thinking, so I suggest something like `link(String name, Entity target)` or perhaps some overloaded version also with the properties.